modules/account-tokenizer/src/lib.rs

```diff
@@ -0,0 +1,498 @@
+// This file is part of Acala.
+
+// Copyright (C) 2020-2022 Acala Foundation.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! # Account Tokenized module
+//!
+//! This module allows Accounts on the Relaychain to be "tokenized" into a
+//! Account Token, in the form of a NFT. The overall workflow is as follows:
+//! 1. User creates an anonymous account using the `Proxy` pallet
+//! 2. User transfers the control of the account to our Parachain account
+//! 3. `request_mint` is called. Foreign state oracles will confirm the mint request
+//! 4. An account token NFT is minted. The token is transferrable.
+//! 5. The owner of the NFT can call `request_redeem` to redeem the token. This will cause
+//!    the transfer of ownership of the anonymous account from the Parachain's account
+//!    to the user's nominated account.
+//! 6. Once the transfer is completed and confirmed by the Oracle, the NFT token is burned.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::unused_unit)]
+#![allow(clippy::upper_case_acronyms)]
+
+use frame_support::{
+	dispatch::{Dispatchable, GetDispatchInfo},
+	log,
+	pallet_prelude::*,
+	require_transactional,
+	traits::{
+		tokens::nonfungibles::{Create, Inspect, Mutate, Transfer},
+		BalanceStatus, Currency,
+		ExistenceRequirement::KeepAlive,
+		GetStorageVersion, NamedReservableCurrency, StorageVersion, WithdrawReasons,
+	},
+	transactional, PalletId,
+};
+
+use frame_system::pallet_prelude::*;
+use orml_traits::{arithmetic::Zero, InspectExtended};
+use sp_io::hashing::blake2_256;
+use sp_runtime::traits::{AccountIdConversion, One, TrailingZeroInput};
+use sp_std::vec::Vec;
+
+use module_support::{CreateExtended, ForeignChainStateQuery, ProxyXcm};
+use primitives::{
+	nft::{ClassProperty, Properties},
+	Balance, ReserveIdentifier,
+};
+
+pub const RESERVE_ID: ReserveIdentifier = ReserveIdentifier::AccountTokenizer;
+// Represents `ProxyType::Any` on relaychain.
+pub const PROXYTYPE_ANY: [u8; 1] = [0_u8];
+
+mod mock;
+mod tests;
+pub mod weights;
+
+pub use module::*;
+pub use weights::WeightInfo;
+
+#[frame_support::pallet]
+pub mod module {
+
+	use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + orml_nft::Config {
```

Comment:
```suggestion
	pub trait Config: frame_system::Config {
```

Shouldn't need it as we have `NFTInterface`

Reply:
We don't need to be generic here. So we can just pul the TokenId/ClassId type from primitives and `Inspect<Self::AccountId, ClassId = ClassId, InstanceId = TokenId>`

Similar to how we handle `Balance` type in other pallets.